subprojects/architecture-test/src/changes/archunit_store/stored.rules

```diff
@@ -1,10 +1,11 @@
 #
-#Wed Aug 02 18:29:12 CEST 2023
+#Wed Dec 13 16:34:21 WET 2023
 methods\ that\ are\ mutable\ public\ API\ properties\ and\ do\ not\ have\ raw\ return\ type\ org.gradle.api.resources.TextResource\ and\ do\ not\ have\ raw\ return\ type\ assignable\ to\ org.gradle.api.file.FileCollection\ should\ have\ return\ type\ Provider=public-api-mutable-properties.txt
 classes\ that\ are\ Gradle\ public\ API\ should\ not\ have\ direct\ super-class\ or\ interface\ that\ are\ Gradle\ Internal\ API=public-api-not-extends-internal-types.txt
 methods\ that\ are\ public\ API\ methods\ should\ have\ only\ arguments\ or\ return\ types\ that\ are\ Gradle\ public\ API\ or\ primitive\ or\ built-in\ JDK\ classes\ or\ Groovy\ classes\ or\ Kotlin\ classes=public-api-methods-return-allowed-types.txt
-methods\ that\ are\ mutable\ public\ API\ properties\ should\ not\ have\ raw\ return\ type\ org.gradle.api.resources.TextResource=public-api-mutable-text-resource.txt
+classes\ that\ is\ injected\ by\ a\ getter\ marked\ with\ @Inject\ or\ is\ injected\ by\ a\ constructor\ marked\ with\ @Inject\ should\ be\ annotated\ with\ @ServiceScope=dependency-injection-service-scope.txt
 no\ classes\ should\ call\ method\ RoleBasedConfigurationContainerInternal.resolvableDependencyScopeUnlocked(String)\ or\ should\ call\ method\ RoleBasedConfigurationContainerInternal.resolvableDependencyScopeUnlocked(String,\ Action)\ or\ should\ call\ method\ RoleBasedConfigurationContainerInternal.maybeCreateResolvableUnlocked(String)\ or\ should\ call\ method\ RoleBasedConfigurationContainerInternal.maybeCreateConsumableUnlocked(String)\ or\ should\ call\ method\ RoleBasedConfigurationContainerInternal.maybeCreateDependencyScopeUnlocked(String)\ or\ should\ call\ method\ RoleBasedConfigurationContainerInternal.maybeCreateDependencyScopeUnlocked(String,\ boolean)\ or\ should\ call\ method\ RoleBasedConfigurationContainerInternal.maybeCreateMigratingUnlocked(String,\ ConfigurationRole)\ or\ should\ call\ method\ RoleBasedConfigurationContainerInternal.maybeCreateResolvableDependencyScopeUnlocked(String),\ because\ Resolvable\ +\ dependency\ scope\ configurations\ should\ be\ avoided\ and\ we\ are\ migrating\ away\ from\ maybeCreate\ methods=configuration-factory-methods.txt
+methods\ that\ are\ mutable\ public\ API\ properties\ should\ not\ have\ raw\ return\ type\ org.gradle.api.resources.TextResource=public-api-mutable-text-resource.txt
```

Comment:
❌  The last time @wolfs asked me to move this line to its original position

Reply:
Moved, verified that refreeze doesn't cause a diff.

Reply:
Okay, when you have a new rule, it _does_ move.
@wolfs WDYT? I understand that it looks better this way, but we are going to have a perpetual battle keeping this order when we introduce new rules.

Reply:
I don't feel strongly about the placement of the rules in the properties file. Though I don't think the order is deterministic, maybe we can/should make the order deterministic somehow?

Reply:
> Though I don't think the order is deterministic

I've looked at the implementation [here](https://github.com/TNG/ArchUnit/blob/2e0ab7ca24290a44130922b8200dd061c29fe8ca/archunit/src/main/java/com/tngtech/archunit/library/freeze/TextFileBasedViolationStore.java#L141), but I haven't found the new rule code path (which is our concern, because at that point it does order)

Reply:
I think the actual writing of the properties file happens [here](https://github.com/TNG/ArchUnit/blob/2e0ab7ca24290a44130922b8200dd061c29fe8ca/archunit/src/main/java/com/tngtech/archunit/library/freeze/TextFileBasedViolationStore.java#L258-L264). That uses `java.util.Properties.store` to write out the file. `Properties` use a hash based datastructure, so it has no ordering.

We could sort the properties in the file in a `doLast` when we re-freeze to keep the order consistent.